### PR TITLE
FIX-#3368: support unsigned integers in OmniSci backend

### LINKED ISSUE #3368: RuntimeError: uint8 is not yet supported.
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: `Ubuntu 20.04`
- **Modin version** (`modin.__version__`): `0.10.2` (from conda-forge)
- **Python version**: `3.9.6`
- **Code we can use to reproduce**:

```shell
conda install modin-omnisci -c conda-forge
MODIN_BACKEND=omnisci MODIN_EXPERIMENTAL=true python test.py
```

Test.py:
```python
import numpy as np
import modin.pandas as pd

test = np.array([1., 3., 5.])
pd.get_dummies(np.array([1., 3., 5.])).sum(axis=0)
```

### Describe the problem

This issue was found when running `PlastiCC` workload.

### Source code / logs

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../modin_on_omnisci_conda_forge/lib/python3.9/site-packages/modin/pandas/dataframe.py", line 2070, in sum
    data._query_compiler.sum(
  File ".../modin_on_omnisci_conda_forge/lib/python3.9/site-packages/modin/experimental/backends/omnisci/query_compiler.py", line 96, in method_wrapper
    return method(self, *args, **kwargs)
  File ".../modin_on_omnisci_conda_forge/lib/python3.9/site-packages/modin/experimental/backends/omnisci/query_compiler.py", line 361, in sum
    return self._agg("sum", **kwargs)
  File ".../modin_on_omnisci_conda_forge/lib/python3.9/site-packages/modin/experimental/backends/omnisci/query_compiler.py", line 96, in method_wrapper
    return method(self, *args, **kwargs)
  File ".../modin_on_omnisci_conda_forge/lib/python3.9/site-packages/modin/experimental/backends/omnisci/query_compiler.py", line 413, in _agg
    new_frame = new_frame._set_index(
  File ".../modin_on_omnisci_conda_forge/lib/python3.9/site-packages/modin/experimental/engines/omnisci_on_ray/frame/data.py", line 1931, in _set_index
    self._execute()
  File ".../modin_on_omnisci_conda_forge/lib/python3.9/site-packages/modin/experimental/engines/omnisci_on_ray/frame/data.py", line 1708, in _execute
    new_partitions = self._partition_mgr_cls.run_exec_plan(
  File ".../modin_on_omnisci_conda_forge/lib/python3.9/site-packages/modin/experimental/engines/omnisci_on_ray/frame/partition_manager.py", line 248, in run_exec_plan
    p.frame_id = omniSession.put_arrow_to_omnisci(obj)
  File ".../modin_on_omnisci_conda_forge/lib/python3.9/site-packages/modin/experimental/engines/omnisci_on_ray/frame/omnisci_worker.py", line 197, in put_arrow_to_omnisci
    cls._server.importArrowTable(name, table, fragment_size=fragment_size)
  File "dbe.pyx", line 207, in omniscidbe.PyDbEngine.importArrowTable
RuntimeError: uint8 is not yet supported.
```

### PR DESCRIPTION
**Pull request by dchigarev**
## What do these changes do?

This PR introduces a native workaround for working with unsigned integers via OmniSci backend. Originally, OmniSci doesn't support unsigned integers, the PR adds a logic of upcasting such types to the compatible ones prior importing them to OmniSci.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3368 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/test/utils.py [hunk 1]
```diff
@@ -178,6 +179,32 @@ def __init__(self, *dfs):
...
+            modin_frame = df._query_compiler._modin_frame
+            # Forcibly executing plan via OmniSci
+            new_partitions = modin_frame._partition_mgr_cls.run_exec_plan(
+                modin_frame._op,
+                modin_frame._index_cols,
+                modin_frame._dtypes,
+                modin_frame._table_cols,
+            )
+            modin_frame._partitions = new_partitions
+            modin_frame._op = FrameNode(modin_frame)
```
[the remark this answers is not in the document]
Comment (vnlitvinov) [reply modin/experimental/core/execution/native/implementations/omnisci_on_native/test/utils.py:204]: This should be a comment in the code. No one would see this comment when the PR is merged.

**Review by vnlitvinov — COMMENTED** [review]
Please embed the comment, otherwise LGTM

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/test/utils.py [hunk 2]
```diff
@@ -178,6 +179,32 @@ def __init__(self, *dfs):
...
+        -------
+        list of modin DataFrames
+            A list of DataFrames whose payload is ``pyarrow.Table`` that was just exported from OmniSci.
+        """
+        result = []
+        for df, frame_id in self._imported_frames:
+            # Append `TransformNode`` selecting all the columns (SELECT * FROM frame_id)
+            df = df[df.columns.tolist()]
+            modin_frame = df._query_compiler._modin_frame
+            # Forcibly executing plan via OmniSci
```
Comment (dchigarev) [inline modin/experimental/core/execution/native/implementations/omnisci_on_native/test/utils.py:196]:
```suggestion
            # Forcibly executing plan via OmniSci. We can't use `modin_frame._execute()` here
            # as it has a chance of running via pyarrow bypassing OmniSci
```

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/test/utils.py [hunk 3]
```diff
@@ -178,6 +179,33 @@ def __init__(self, *dfs):
     def __enter__(self):
         return self
 
+    def export_frames(self):
+        """
+        Export tables from OmniSci that was imported by this instance.
+
+        Returns
+        -------
+        list of modin DataFrames
```
Comment (YarShev) [inline modin/experimental/core/execution/native/implementations/omnisci_on_native/test/utils.py:188]:
```suggestion
        list
```

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/test/utils.py [hunk 4]
```diff
@@ -178,6 +179,33 @@ def __init__(self, *dfs):
...
+            A list of DataFrames whose payload is ``pyarrow.Table`` that was just exported from OmniSci.
```
Comment (YarShev) [inline modin/experimental/core/execution/native/implementations/omnisci_on_native/test/utils.py:189]:
```suggestion
            A list of Modin DataFrames whose payload is ``pyarrow.Table``
            that was just exported from OmniSci.
```

**Review by YarShev — APPROVED** [review]

**Comment by dchigarev** [comment]
@vnlitvinov @YarShev the comments were addressed and CI is green

**Review by vnlitvinov — APPROVED** [review]
LGTM, thanks!

**Pull request merged**